src/middlewares/likeDislikeCheck.js

```diff
@@ -0,0 +1,60 @@
+import models from '../api/models';
+import statusCodes from '../helpers/constants/status.codes';
+
+/**
+ * @class checkLikesDislikes
+ * @description a class to check if a user has already like or disliked a comment
+ */
+export default class checkLikesDislikes {
+  /**
+   * Verify if the user has already liked the comment
+   * @param {Object} req - Request
+   * @param {Object} res  - Response
+   * @param {Function} next -Next
+   * @returns {Object} The response object
+   */
+  static async liked(req, res, next) {
+    const { commentId } = req.params;
+    const { id } = req.user.user;
+    const hasLiked = await models.LikeDislike.findAll({
```

Comment:
I think there is some repetitiveness between your middleware and your controllers, so my suggestion would be, why don't you put all this logic `` const hasDisliked = await models.LikeDislike.findAll({
      where: {
        commentId,
        userId: id,
        dislikes: 1,
        likes: 0
      }
    });
    if (hasDisliked[0]) {
      await models.LikeDislike.update({ dislikes: 0, likes: 1 },
        { where: { id: hasDisliked[0].id } });
      return res.status(statusCode.OK).json({
        message: 'Thank you for reacting on  this comment!'
      });
    }`` inside a middleware to check if the user has disliked a comment and leave the rest in your controller

Reply:
Correct if I am wrong @murediane

Reply:
The middleware was just to reduce complexity.I am checking if the action was node done before . It is normal that some things may look similar. Coz I am checking the actions that are being performed in the controller. But the middleware and controllers are doing different things. The action to like for example can need to create if it is the first time the user liked or update if the user has disliked before. in the middleware, I am just checking if the user trying to like has liked before or not. Still, I need to check that in the controller to know which transaction to do whether it is create or update.Does that make sense to you